src/autora/experimentalist/random_.py

```diff
@@ -0,0 +1,341 @@
+"""Tools to make randomly sampled experimental conditions."""
+import random
+from functools import singledispatch
+from typing import Optional, Union
+
+import numpy as np
+import pandas as pd
+
+from autora.state.delta import Result, State, wrap_to_use_state
+from autora.variable import ValueType, VariableCollection
+
+
+@singledispatch
+def random_pool(s, **kwargs):
+    """Function to create a sequence of conditions randomly sampled from independent variables."""
+    raise NotImplementedError(
+        "random_pool doesn't have an implementation for %s (type=%s)" % (s, type(s))
+    )
+
+
+@random_pool.register(State)
+def random_pool_on_state(
+    s: State,
+    num_samples: int = 5,
+    random_state: Optional[int] = None,
+    replace: bool = True,
+) -> State:
+    """
+
+    Args:
+        variables:
+        fmt: the output type required
+
+    Returns:
```

Comment:
i've noticed that occasionally certain entries are left blank in docstrings (not just here, but generally); is this an intentional/reasoned in certain circumstances?

Reply:
no, pure failure to notice that they were missing. I'll update those. Thanks!